### PR TITLE
RMN config integration

### PR DESCRIPTION
**Pull request by 0xnogo**
* Use of RMN home reader + remote configs inside the rmn controller
* Propagating the RMNRemote config to the Outome in report generation phase
* Use the ReportWithInfo's `Info` to share the minSigners to the `ShouldAcceptAttestedReport`

### CODE REVIEW

File: commit/report.go [hunk 1]
```diff
@@ -45,7 +50,18 @@ func (p *Plugin) Reports(seqNr uint64, outcomeBytes ocr3types.Outcome) ([]ocr3ty
...
+		MinSigners: outcome.MerkleRootOutcome.RMNRemoteCfg.MinSigners,
+	}
+
+	// Serialize reportInfo to []byte
+	infoBytes, err := json.Marshal(reportInfo)
+	if err != nil {
+		return nil, fmt.Errorf("failed to marshal report info: %w", err)
+	}
+
+	return []ocr3types.ReportWithInfo[[]byte]{{Report: encodedReport, Info: infoBytes}}, nil
```
Comment (0xnogo) [inline commit/report.go:64]: Leverage the Info to access the `MinSigners` in `ShouldAcceptAttestedReport`

File: commit/merkleroot/outcome.go [hunk 2]
```diff
@@ -188,6 +188,7 @@ func buildReport(
 		RootsToReport:       roots,
 		OffRampNextSeqNums:  prevOutcome.OffRampNextSeqNums,
 		RMNReportSignatures: sigs,
+		RMNRemoteCfg:        prevOutcome.RMNRemoteCfg,
```
Comment (makramkd) [inline commit/merkleroot/outcome.go:191]: Shouldn't we come to consensus on this prior to including it in the outcome?

Comment (0xnogo) [reply commit/merkleroot/outcome.go:191]: The consensus is reached in another round where state is `SelectingRangesForReport`. https://github.com/smartcontractkit/chainlink-ccip/blob/febfb02821d8ba19477958cafec609bc1774b1b2/commit/merkleroot/outcome.go#L52

There we will gather all the AO and reach consensus.

Here we are only propagating the RMNRemote config.

Comment (makramkd) [reply commit/merkleroot/outcome.go:191]: Oh I see, yeah that makes sense 👍

File: commit/report.go (hunk 1 above)
Comment (0xnogo) [reply commit/report.go:64]: I checked with research team and it seems like this use case fits the usage the `ReportWithInfo`

File: commit/factory.go [hunk 3]
```diff
@@ -18,7 +18,6 @@ import (
 	cciptypes "github.com/smartcontractkit/chainlink-common/pkg/types/ccipocr3"
```
[the remark this answers is not in the document]
Comment (0xnogo) [reply commit/factory.go:18]: As agreed offline, this is going to be fixed in a later PR.

File: commit/merkleroot/rmn/controller.go [hunk 4]
```diff
@@ -769,30 +823,48 @@ func (c *controller) validateReportSigResponse(
...
 	return &reportSigWithSignerAddress{
 		reportSig:     reportSig,
-		signerAddress: rmnNode.SignReportsAddress,
+		signerAddress: signerNode.OnchainPublicKey,
 	}, nil
 }
 
-func (c *controller) getRmnNodeByID(nodeID NodeID) (RMNNodeInfo, bool) {
-	for _, node := range c.rmnCfg.Home.RmnNodes {
+func (c *controller) getHomeNodeByID(
```
[the remark this answers is not in the document]
Comment (0xnogo) [reply commit/merkleroot/rmn/controller.go:830]: 100% agree

File: commit/report.go [hunk 5]
```diff
@@ -45,7 +50,18 @@ func (p *Plugin) Reports(seqNr uint64, outcomeBytes ocr3types.Outcome) ([]ocr3ty
...
 	}
 
-	return []ocr3types.ReportWithInfo[[]byte]{{Report: encodedReport, Info: nil}}, nil
+	// Prepare the info data
+	reportInfo := ReportInfo{
+		MinSigners: outcome.MerkleRootOutcome.RMNRemoteCfg.MinSigners,
+	}
+
+	// Serialize reportInfo to []byte
+	infoBytes, err := json.Marshal(reportInfo)
```
[the remark this answers is not in the document]
Comment (0xnogo) [reply commit/report.go:59]: Good point. I added it in the report.go for now. But I will port that to the chainlink-common

**Pull request merged**